internal/executor/job/job_context.go

```diff
@@ -287,3 +295,22 @@ func (c *ClusterJobContext) handleDeletedPod(pod *v1.Pod) {
 		}
 	}
 }
+
+func (c *ClusterJobContext) checkForPreemption(pod *v1.Pod, jobId string) bool {
+	events := c.clusterContext.GetEvents()
+	for _, e := range events {
+		isCurrentDeletedPod := e.InvolvedObject.Name == pod.Name
+		if util.IsPreemptedEvent(e) && isCurrentDeletedPod {
+			c.registerIssue(jobId, &PodIssue{
```

Comment:
I'd move this to the caller. So we don't have two different functions that may register an issue.

Reply:
So the caller would be something like 
```
if preemptionEvent := c.checkForPreemption(pod, jobId); preemptionEvent {
// register preemption issue
} else {
// register ExternallyDeleted issue
}
```